### PR TITLE
improve sharing autocomplete tests

### LINKED ISSUE #1473: all users and groups that contain the string {string} in their name should be listed in the autocomplete list on the webUI does not do what it promisses
the test code of `Then all users and groups that contain the string {string} in their name should be listed in the autocomplete list on the webUI` only checks if every item in the list contains the typed string, but it should test if all users that are expected to be listed, are listed

### PR DESCRIPTION
**Pull request by individual-it**
## Description
1. remember created groups
2. delete groups with `DELETE` not `POST` :see_no_evil: 
3. make the step `Then all users and groups that contain the string {string} in their name should be listed in the autocomplete list on the webUI` check if all users and groups, that are expected to be in the list, are in the list

## Related Issue

- Fixes #1473

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4043/10

<p>






</p></details>

File: tests/acceptance/helpers/userSettings.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ module.exports = {
     }
   },
   createdUsers: {},
+  createdGroups: {},
```
Comment (paurakhsharma) [inline tests/acceptance/helpers/userSettings.js]: Maybe it's better we just create `createdGroups` as an array.
Later if we need other attributes in a created group we can add it.

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 2]
```diff
@@ -91,9 +91,39 @@ When('the user shares file/folder {string} with user {string} using the webUI',
...
+          let groupString = groupId + client.page.FilesPageElement.sharingDialog().getGroupSharePostfix()
+
+          assert.ok(
+            itemsList.includes(groupString),
+            `could not find '${groupString}' in autocomple share list`
+          )
+        }
+      }
+
+      // check if every listed item contains the pattern
```
Comment (paurakhsharma) [inline tests/acceptance/stepDefinitions/sharingContext.js]: I don't think this logic should be in this step.
We can create a different step which uses this logic.

**Comment by individual-it** [comment]
@paurakhsharma pleas review again

**Pull request merged**